lib/ansible/modules/network/cnos/cnos_vlan.py

```diff
@@ -0,0 +1,308 @@
+     arguments, see [Overloaded Variables].
+     This module uses SSH to manage network device configuration.
+     The results of the operation can be viewed in results directory.
+     To know more about this module from Lenovo and customizing them for your
+     use cases, please visit our [User Guide](http://systemx.lenovofiles.com/
+     help/index.jsp?topic=%2Fcom.lenovo.switchmgt.ansible.doc%2Fansible_for_
+     cnos.html)
+version_added: "2.3"
+Options:
+
+Overloaded Options:
+    - The following is a table depicting how the overloaded variables are used
+     in the context of VLAN.
+    - vlanArg1 is required
+    - Other variables are specific to the CLI command defined in vlanArg1
+    - The words in bold are CLI command parameters. When confronted with a
+     list of bulleted options (•), you must choose only one to use.
+    - Please refer to the CNOS Command Reference specific to the switch you
+     are configuring for details of the CLI commands and parameters.
+vlanArg1: [access-map,access-map,access-map,dot1q,filter,<1-3999> VLAN ID
+1-3999 or range,<1-3999> VLAN ID 1-3999 or range,<1-3999> VLAN ID 1-3999 or
+range,<1-3999> VLAN ID 1-3999 or range,<1-3999> VLAN ID 1-3999 or range,
+<1-3999> VLAN ID 1-3999 or range,<1-3999> VLAN ID 1-3999 or range,<1-3999>
+VLAN ID 1-3999 or range,<1-3999> VLAN ID 1-3999 or range,<1-3999> VLAN ID
+1-3999 or range,<1-3999> VLAN ID 1-3999 or range,<1-3999> VLAN ID 1-3999 or
+range,<1-3999> VLAN ID 1-3999 or range,<1-3999> VLAN ID 1-3999 or range,
+<1-3999> VLAN ID 1-3999 or range,<1-3999> VLAN ID 1-3999 or range]
+vlanArg2: [VLAN Access Map name,VLAN Access Map name,VLAN Access Map name,
+egress-only,VLAN Access Map Name,name,flood,state,ip,ip,ip,ip,ip,ip,ip,ip,ip,
+ip,ip,ip,ip]
+vlagArg3: [drop or forward or redirect,ip or mac,~,~,~,~,~,~,~, Interval in
+seconds,ethernet,port-aggregation,Querier IP address,Querier Timeout in
+seconds,Query Interval in seconds, Query Max Response Time in seconds,~,
+Robustness Variable value, Number of queries sent at startup,Query Interval
+at startup,~]
+vlagArg4: [~,access-list name,~,~,~,~,~,~,~,~,Slot/chassis number,Port
+Aggregation Number,~,~,~,~,~,~,~,~,~]
+Remarks: [~,~,~,vlanArg2 is optional,~,~,vlanArg3 is optional,~,~,~,~,~,~,~,~,
+~,~,~,~,~,This feature is not supported. Please use runcommand option]
+
+Notes:
+    - For help in developing on modules, should you be so inclined, please read
+     [Community Information & Contributing](http://docs.ansible.com/ansible/
+     community.html) , [Helping Testing PRs](http://docs.ansible.com/ansible/
+     dev_guide/developing_test_pr.html),[and Developing Modules]
+     (http://docs.ansible.com/ansible/dev_guide/developing_modules.html).
+
+Module Dependency :
+    - cnos_vlan.py
+    - cnos.py
+
+'''
+EXAMPLES = '''
+
+Tasks: The following are examples of using the module cnos_vlan. These are written in the main.yml file of the tasks directory.
+---
+- name: Test Vlan - Create a vlan, name it
```

Comment:
Please use multi line yaml with a `- name:`

Please see http://docs.ansible.com/ansible/dev_guide/developing_modules_documenting.html#examples-block for details